lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1808,18 +1808,128 @@ bool SwiftLanguageRuntimeImpl::IsValidErrorValue(ValueObject &in_value) {
 
   return true;
 }
+/// Digs into the protocols inside a protocol composition and checks if 
+/// any of them belong to the LLDB module.
+static bool
+IsDeclaredInLLDBExpr(const swift::reflection::TypeRef *typeref) {
+  using swift::Demangle::Node;
+
+  const auto *protocol_typeref = llvm::dyn_cast_or_null<
+      const swift::reflection::ProtocolCompositionTypeRef>(typeref);
+
+  if (!protocol_typeref)
+    return false;
+
+  swift::Demangle::Demangler dem;
+
+  NodePointer type = protocol_typeref->getDemangling(dem);
```

Comment:
For a followup-commit: We should factor this out into a static helper and unit-test it.

Reply:
You mean the whole method?

Reply:
The remainder of the method that operates on the demangle tree. It doesn't need a live process or reflection metadata and can be easily unit-tested, so we should do that :-)

Reply:
Aah ok!